src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,306 @@
+    ) {
+      focusAsync(this._previouslyFocusedElementOutsideTrapZone)
+    }
+
+    this._unsubscribeFromEvents()
+    const lastActiveFocusTrap = FocusTrapZone._focusStack[FocusTrapZone._focusStack.length - 1]
+
+    if (!FocusTrapZone._focusStack.length) {
+      this._showContentInAccessibilityTree()
+    } else if (
+      lastActiveFocusTrap._root.current &&
+      lastActiveFocusTrap._root.current.hasAttribute(HIDDEN_FROM_ACC_TREE)
+    ) {
+      lastActiveFocusTrap._root.current.removeAttribute(HIDDEN_FROM_ACC_TREE)
+      lastActiveFocusTrap._root.current.removeAttribute('aria-hidden')
+    }
+  }
+
+  public focus() {
+    if (!this._root.current) return
+    const { focusPreviouslyFocusedInnerElement, firstFocusableSelector } = this.props
+
+    if (
+      focusPreviouslyFocusedInnerElement &&
+      this._previouslyFocusedElementInTrapZone &&
+      this._root.current.contains(this._previouslyFocusedElementInTrapZone)
+    ) {
+      // focus on the last item that had focus in the zone before we left the zone
+      focusAsync(this._previouslyFocusedElementInTrapZone)
+      return
+    }
+
+    const focusSelector =
+      typeof firstFocusableSelector === 'string'
+        ? firstFocusableSelector
+        : firstFocusableSelector && firstFocusableSelector()
+
+    let _firstFocusableChild
+
+    if (focusSelector) {
+      _firstFocusableChild = this._root.current.querySelector('.' + focusSelector)
+    } else {
+      _firstFocusableChild = getNextElement(
+        this._root.current,
+        this._root.current.firstChild as HTMLElement,
+        true,
+        false,
+        false,
+        true,
+      )
+    }
+
+    _firstFocusableChild && focusAsync(_firstFocusableChild)
+  }
+
+  private _onFocusCapture = (ev: React.FocusEvent<HTMLDivElement>) => {
+    this.props.onFocusCapture && this.props.onFocusCapture(ev)
+    if (ev.target !== ev.currentTarget) {
+      // every time focus changes within the trap zone, remember the focused element so that
+      // it can be restored if focus leaves the pane and returns via keystroke (i.e. via a call to this.focus(true))
+      this._previouslyFocusedElementInTrapZone = ev.target as HTMLElement
+    }
+  }
+
+  private _onKeyboardHandler = (ev: React.KeyboardEvent<HTMLDivElement>): void => {
+    this.props.onKeyDown && this.props.onKeyDown(ev)
+
+    if (
+      ev.isDefaultPrevented() ||
+      keyboardKey.getCode(ev) !== keyboardKey.Tab ||
+      !this._root.current
+    ) {
+      return
+    }
+
+    const _firstTabbableChild = getFirstTabbable(
+      this._root.current,
+      this._root.current.firstChild as HTMLElement,
+      true,
+    )
+    const _lastTabbableChild = getLastTabbable(
+      this._root.current,
+      this._root.current.lastChild as HTMLElement,
+      true,
+    )
+
+    if (ev.shiftKey && _firstTabbableChild === ev.target) {
+      focusAsync(_lastTabbableChild)
+      ev.preventDefault()
```

Comment:
these two lines are the same for both logical branches - lets not duplicate them and put out of scope of `if-else`